liby2util-r/src/MemUsage.cc

```diff
@@ -51,11 +51,12 @@ void MemUsage::MuDump ()
 	ie = m_mu_instances->end ();
     for (; ii != ie; ++ii)
     {
-	const char * name = typeid (**ii).name ();
+	MemUsage * tracked_ptr = *ii;
+	const char * name = typeid (*tracked_ptr).name ();
```

Comment:
Dereferencing the iterator is an operation with side effects and Clang warns when such an operation is inside a `typeid`. Factoring it out prevents the warning.